### ISSUE TITLE
Theme "Advanced" Usage ... Under Construction.

### ISSUE THREAD

**Issue by nelsonic** [dwyl/nextra-demo#7]
Trying to customise the theme of our `Nextra` site to change the `<footer>` which defaults to: 
"MIT 2024 © Nextra."

<img width="1027" alt="nextra-footer-default" src="https://github.com/user-attachments/assets/ecb1f054-3ce6-4e76-a1da-7dd9dbf74a40">

Reading the `Footer` docs: https://nextra.site/docs/docs-theme/theme-configuration#footer

<img width="1025" alt="image" src="https://github.com/user-attachments/assets/8dd0bef1-5f40-4cde-8b3f-363598c8b68c">

```jsx
export default {
  footer: {
    text: (
      <span>
        MIT {new Date().getFullYear()} ©{' '}
        <a href="https://nextra.site" target="_blank">
          Nextra
        </a>
        .
      </span>
    )
  }
}
```

<img width="1027" alt="nextra-footer-updated" src="https://github.com/user-attachments/assets/cab20c4f-a83d-4c11-be9b-f1e823a60a3e">

But given that this is a `<text>` attribute, `Nextra` is still controlling the appearance in the `<footer>`.
This is the markup that is rendered:

<img width="1038" alt="nextra-footer-markup" src="https://github.com/user-attachments/assets/34549ece-48bf-4481-94ce-3a3c5614110a">

What I want is to _control_ the `<footer>` _completely_. I don't just want to update the **`<text>`** ...
So I went back to the `Nextra` docs to try and use a `<Component>` for the `<footer>` ... nothing. 🤷‍♂️ 😢 
Googled for a bit and found: https://github.com/shuding/nextra/issues/2926#issuecomment-2162909549
<img width="821" alt="image" src="https://github.com/user-attachments/assets/22bc9c54-76a9-4607-9e62-a5082ab92730">

## "_You might have to experiment with specific css selectors_" ... 

He links to: https://nextra.site/docs/custom-theme#advanced-usage
<img width="1172" alt="advanced-usage-under-construction" src="https://github.com/user-attachments/assets/9e39d63a-3888-41d5-b72f-af6265f2591c">

Under construction ... 🚧 
This is such a time-suck! 😢 
I've spent _waaaay_ too much time on this! ⏳ 
just want a good-looking `<footer>` ... is that too much to ask?! 🤷‍♂️ 
I _tried_ the `CSS` selectors approach and it was _hell_ 😡 
I don't want to have to use 50 **`important!`** statements 
to _attempt_ to override the inline `Tailwind` styles ... it's _horrible_! 🤦‍♂️ 
Just going to have to leave it _ugly_ for now. 🙈 
And _maybe_ `return` to this later. ⏳

**Comment by nelsonic** [comment]
Tried creating a custom component but `Nextra` didn't like it:

<img width="923" alt="image" src="https://github.com/user-attachments/assets/81936f1b-1405-4542-8d78-c4d0339e2c8f">

```sh
Module parse failed: Unexpected token (2:9)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export default function footerComponent() {
>   return <h1>Test</h1>;
| }
|
```

**Comment by anuragmitra4** [comment]
After some framework fighting and looking at [some source code](https://github.dev/shuding/nextra/blob/main/packages/nextra-theme-docs/src/components/navbar.tsx), seems like the way to customize the footer (or any component) is to use tailwind classes with a `nx-` prefix. 

As an example, I tried to update the [`project.icon`](https://nextra.site/docs/docs-theme/theme-configuration#project-link) component:
```
import { Avatar, AvatarImage } from "./components/ui/avatar";
import { Button } from "./components/ui/button";
import { ChevronDown } from "lucide-react";

const config = {
  project: {
    link: "https://github.com/shadcn.png",
    icon: (
      <Button className="nx-flex nx-items-center">
        <Avatar className="nx-h-8 nx-h-8 nx-relative nx-flex nx-shrink-0 nx-overflow-hidden nx-rounded-full">
          <AvatarImage src="https://github.com/shadcn.png" alt="@shadcn" />
        </Avatar>

        <p className="nx-flex nx-flex-col nx-ml-2 nx-text-left nx-text-xs sm:nx-block hidden">
          <strong className="block font-medium">Shadcn</strong>
          <span className="text-gray-500">shadcnui.com</span>
        </p>

        <ChevronDown className="nx-m-0.5 nx-h-4 nx-w-4" />
      </Button>
    ),
  },
};

export default config;
```
to get
<img width="478" alt="Screenshot 2024-08-21 at 4 59 50 PM" src="https://github.com/user-attachments/assets/219f38c4-7ded-4700-a9fa-03bb03067921">

**Comment by anuragmitra4** [comment]
Update: After doing some more testing, the situation is made worse by the fact that tailwind only packages the classes being used by the nextra-docs-theme package. So, if the package has never referenced `nx-bg-gray-700`, you will not be able to use that class using tailwind. Quite frustrating.

**Comment by nelsonic** [comment]
Beyond frustrating. Makes any customisation take _way_ longer than it should. ⏳

**Comment by nelsonic** [comment]
Yeah, the `nx-` prefix for all `TailwindCSS` classes is a horrible choice for extensibility 🙅 
makes reusing components from the "frontend" (`Next.js` App) _extra_ difficult
as we have to _manually_ add the prefix or load a second instance of `Tailwind`. 🙃
